tests/integration/vxlan_test.py

```diff
@@ -0,0 +1,118 @@
+from libnmstate.schema import InterfaceType
+
+from .testlib import assertlib
+from .testlib.vxlan import apply_state
+from .testlib.vxlan import vxlan_down
+from .testlib.vxlan import vxlan_interfaces
+from .testlib.vxlan import vxlan_up
+from .testlib.vxlan import vxlans_absent
+from .testlib.vxlan import vxlans_up
+
+VXLAN1_ID = 101
+VXLAN2_ID = 102
+
+
+def test_add_and_remove_vxlan(eth1_up):
+    ifname = eth1_up[Interface.KEY][0][Interface.NAME]
+    with vxlan_interfaces(
+        [(VXLAN1_ID, ifname, '192.168.100.1')]
+    ) as desired_state:
+        assertlib.assert_state(desired_state)
+
+    vxlan1_ifname = desired_state[Interface.KEY][0][Interface.NAME]
+    assertlib.assert_absent(vxlan1_ifname)
+
+
+def test_add_and_remove_two_vxlans_on_same_iface(eth1_up):
+    ifname = eth1_up[Interface.KEY][0][Interface.NAME]
+    with vxlan_interfaces(
+        [
+            (VXLAN1_ID, ifname, '192.168.100.1'),
+            (VXLAN2_ID, ifname, '192.168.100.2'),
+        ]
+    ) as desired_state:
+        assertlib.assert_state(desired_state)
+
+    vxlan_interfaces_name = (
+        i[Interface.NAME] for i in desired_state[Interface.KEY]
+    )
+    assertlib.assert_absent(vxlan_interfaces_name)
+
+
+def test_rollback_for_vxlans(eth1_up):
+    ifname = eth1_up[Interface.KEY][0][Interface.NAME]
+    current_state = libnmstate.show()
+    desired_state = {
+        Interface.KEY: [
+            vxlan_up(VXLAN1_ID, ifname, '192.168.100.1'),
+            vxlan_up(VXLAN2_ID, ifname, '192.168.100.2'),
+        ]
+    }
+
+    desired_state[Interface.KEY][1]['invalid_key'] = 'foo'
+    with pytest.raises(NmstateVerificationError):
+        libnmstate.apply(desired_state)
+
+    time.sleep(5)  # Give some time for NetworkManager to rollback
+    current_state_after_apply = libnmstate.show()
+    assert current_state == current_state_after_apply
+
+
+def test_set_vxlan_iface_down(eth1_up):
+    ifname = eth1_up[Interface.KEY][0][Interface.NAME]
+    with vxlan_interfaces(
+        [(VXLAN1_ID, ifname, '192.168.100.1')]
+    ) as desired_state:
+        libnmstate.apply({Interface.KEY: [vxlan_down(VXLAN1_ID, ifname)]})
+
+    vxlan_ifname = desired_state[Interface.KEY][0][Interface.NAME]
+    assertlib.assert_absent(vxlan_ifname)
+
+
+def test_add_new_base_iface_with_vxlan():
+    parent = 'dummy00'
+    up_interfaces = vxlans_up([(VXLAN1_ID, parent, '192.168.100.2')])
+    up_interfaces[Interface.KEY].append(
+        {
+            Interface.NAME: parent,
+            Interface.TYPE: InterfaceType.DUMMY,
+            Interface.STATE: InterfaceState.UP,
+        }
+    )
+    absent_interfaces = vxlans_absent([(VXLAN1_ID, parent)])
+    absent_interfaces[Interface.KEY].append(
+        {
+            Interface.NAME: parent,
+            Interface.TYPE: InterfaceType.DUMMY,
+            Interface.STATE: InterfaceState.ABSENT,
+        }
+    )
+    with apply_state(up_interfaces, absent_interfaces) as desired_state:
```

Comment:
This one seem generic, maybe too generic.
Perhaps move it to this test module, it does not seem to belong to vxlan test helper.

Also, the absent state can be auto-generated from the "up" state by just changing the state.

Reply:
About absent, you may want to just put as absent some of them.